### PR TITLE
Dual frames part 0: rename frames {Inertial, Grid, Logical} to {Physical, LastTimeIndependent, ElementLogical}

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Early refactors for implementing dual frames:
- Rename frames {Inertial, Grid, Logical} to {Physical, LastTimeIndependent, ElementLogical}
- Add frame `BlockLogical` which is not used yet.

These name changes were discussed a few weeks back with @wthrowe and @kidder.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by nilsvu** [comment]
Discussed possible names with @kidder and @nilsdeppe:

`Frame::LastTimeIndependent` -> `Frame::GlobalTimeIndependent`

Suggestions for `Frame::Physical`:
- `Frame::System` (it's the coordinates that the system equations are formulated in, "SystemCoordinates")
- <s>`Frame::Equations`</s> ("EquationsCoordinates" doesn't work)
- <s>`Frame::Cauchy`</s>
- <s>`Frame::Global`</s> (confusion with SpEC 😢)
- <s>`Frame::Primary`</s> (we may have coordinates maps on top of this)
- <s>`Frame::Fundamental`</s>
- <s>`Frame::Principal`</s>

**Comment by nilsdeppe** [comment]
I've pushed a commit that renames LTI->GTI. Squashing this is a bit of a pain because of rebase conflicts so I'd like to avoid that...

**Review by kidder — DISMISSED** [review]

File: docs/GroupDefs.hpp [hunk 1]
```diff
@@ -68,21 +68,21 @@
...
- * Each map under the governance of the control system has an associated
- * time-dependent map parameter \f$\lambda(t)\f$ that is a piecewise Nth order
- * polynomial. At discrete times (called reset times), the control system resets
- * the Nth time derivative of \f$\lambda(t)\f$ to a new constant value, in order
- * to minimize an error function \f$Q(t)\f$ that is specific to each map. At
- * each reset time, the Nth derivative of \f$\lambda(t)\f$ is set to a function
- * \f$U(t)\f$, called the control signal, that is determined by \f$Q(t)\f$ and
- * its time derivatives and time integral. Note that \f$\lambda(t)\f$,
- * \f$U(t)\f$, and \f$Q(t)\f$ can be vectors.
+ * the fixed computational frame (ListTimeIndependent frame) and the physical
```
Comment (nilsvu) [inline docs/GroupDefs.hpp]: typo

File: src/DataStructures/Tensor/IndexType.hpp [hunk 2]
```diff
@@ -39,16 +39,17 @@ namespace Frame {
 /// evaluate an analytic solution in that frame.
 struct FrameIsPhysical {};
 
-struct Logical {};
-struct Grid {};
-struct Inertial : FrameIsPhysical {};
+struct BlockLogical {};
+struct ElementLogical {};
+struct GlobalTimeIndependent {};
+struct Physical : FrameIsPhysical {};
```
Comment (nilsvu) [inline src/DataStructures/Tensor/IndexType.hpp:80]: (This comment is not a change request for this PR.)

Please add a brief documentation what this frame means and why we chose this name; can be in a follow-up PR. We can add more detailed documentation about frames later.

As discussed in person, I'm not happy with the name `Physical` but don't have a better suggestion, so won't oppose this change. I'll keep trying to come up with a better name.

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: docs/GroupDefs.hpp (hunk 1 above)
Comment (nilsdeppe) [reply docs/GroupDefs.hpp]: fixed

File: src/DataStructures/Tensor/IndexType.hpp (hunk 2 above)
Comment (nilsdeppe) [reply src/DataStructures/Tensor/IndexType.hpp:80]: Since there was the typo to fix I pushed a commit with the documentation anyway

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
Rebased on develop

**Review by kidder — APPROVED** [review]

**Review by nilsvu — CHANGES_REQUESTED** [review]
I suggest to rename `Frame::Physical` -> `Frame::System`, since these are the coordinates that the system equations are formulated in, i.e. the "system coordinates".

**Comment by nilsdeppe** [comment]
Since there's no agreement I will develop spectre on my private fork.

Edit:

I've been asked to clarify a few things:
- I don't care about the name.
- I don't want to go through the effort of renaming everything, having another discussion on the name while other PRs are merged, and having to rebuild the whole code base (which takes 20 minutes).

**Comment by nilsdeppe** [comment]
@nilsleiffischer agreed to take over the PR because it is easier to rename "inertial" -> "system" than "physical" -> "system" in terms of the frame. To make life easier I'm closing this PR and he will open a separate PR.

**Pull request closed without merging**